lib/ansible/modules/cloud/scaleway/scaleway_lb_facts.py

```diff
@@ -0,0 +1,130 @@
+options:
+
+  region:
+    description:
+    - Scaleway zone
+    required: true
+    choices:
+      - nl-ams
+      - fr-par
+'''
+
+EXAMPLES = r'''
+- name: Gather Scaleway organizations facts
+  scaleway_lb_facts:
+'''
+
+RETURN = r'''
+---
+scaleway_lb_facts:
+  description: Response from Scaleway API
+  returned: success
+  type: complex
+  contains:
+     "scaleway_lb_facts": [
+                {
+                    "backend_count": 0,
+                    "description": "Load-balancer used for testing scaleway_lb_facts ansible module",
+                    "frontend_count": 0,
+                    "id": "00000000-0000-0000-0000-000000000000",
+                    "instances": [
+                        {
+                            "id": "00000000-0000-0000-0000-000000000000",
+                            "ip_address": "10.14.229.145",
+                            "region": "fr-par",
+                            "status": "ready"
+                        },
+                        {
+                            "id": "00000000-0000-0000-0000-000000000000",
+                            "ip_address": "10.14.208.171",
+                            "region": "fr-par",
+                            "status": "ready"
+                        }
+                    ],
+                    "ip": [
+                        {
+                            "id": "00000000-0000-0000-0000-000000000000",
+                            "ip_address": "51.159.26.123",
+                            "lb_id": "00000000-0000-0000-0000-000000000000",
+                            "organization_id": "00000000-0000-0000-0000-000000000000",
+                            "region": "fr-par",
+                            "reverse": ""
+                        }
+                    ],
+                    "name": "lb_facts_ansible_test",
+                    "organization_id": "00000000-0000-0000-0000-000000000000",
+                    "region": "fr-par",
+                    "status": "ready",
+                    "tags": [
+                        "first_tag",
+                        "second_tag"
+                    ]
+                }
+            ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.scaleway import (
+    Scaleway,
+    ScalewayException,
+    scaleway_argument_spec,
+    SCALEWAY_REGIONS,
+    SCALEWAY_ENDPOINT)
+
+
+class ScalewayLbFacts(Scaleway):
+
+    def __init__(self, module):
+        super(ScalewayLbFacts, self).__init__(module)
+
+        region = module.params["region"]
+        api_version = module.params["api_version"]
+        self.module.params['api_url'] = "/".join([SCALEWAY_ENDPOINT,
+                                                  'lb/%s/regions/%s' % (api_version, region)])
+
+        self.name = 'lbs'
+
+
+def main():
+    argument_spec = scaleway_argument_spec()
+    argument_spec.update(dict(
+        region=dict(required=True, choices=SCALEWAY_REGIONS),
```

Comment:
All Scaleway products are not available in all regions/zones. I think it could be interesting to set the choices explicitly to avoid problems if we add a new region in `SCALEWAY_REGIONS` in which the `lb` product is not available.

Here is a proposal inspired by [what we have implemented in the new Go SDK](https://github.com/scaleway/scaleway-sdk-go/blob/master/utils/locality.go):
```py
# ansible/lib/ansible/module_utils/scaleway.py

SCALEWAY_REGION_FR_PAR = "fr-par"
SCALEWAY_REGION_NL_AMS = "nl-ams"

SCALEWAY_ALL_REGIONS = [
    SCALEWAY_REGION_FR_PAR,
    SCALEWAY_REGION_NL_AMS,
]

SCALEWAY_ZONE_FR_PAR_1 = "fr-par-1"
SCALEWAY_ZONE_FR_PAR_2 = "fr-par-2"
SCALEWAY_ZONE_NL_AMS_1 = "nl-ams-1"

SCALEWAY_ALL_ZONES = [
   SCALEWAY_ZONE_FR_PAR_1,
   SCALEWAY_ZONE_FR_PAR_2,
   SCALEWAY_ZONE_NL_AMS_1,
]
```

```py
# lib/ansible/modules/cloud/scaleway/scaleway_lb_facts.py

def main():
    argument_spec = scaleway_argument_spec()
    argument_spec.update(dict(
        region=dict(required=True, choices=[SCALEWAY_REGION_FR_PAR, SCALEWAY_REGION_NL_AMS]),
    ))
    # ...
```

Reply:
This would imply a change in the ansible code every time a new zone is available. Could we simply pass the value given by the user and see what come out of it? 

I've noticed that name of instances changed, new instances name come and go. Plus in case a new zone come out and an ansible release is not happening, the zone is not usable because the choices are hard-coded and cannot be overwritten with your solution.

Reply:
Yes sorry, there is a misunderstanding; the region `choices` have to be removed (but not the `required` in my opinion). Same for the other products.

Moreover, I think that we can keep the constants for region/zone (cf. my example of `module_utils/scaleway.py` above) and adopt it in the whole project.

Reply:
Added in module utils